File: stlm-java/src/segment/BaseSegment.java
package segment;

public class BaseSegment extends InternalSegment {

	protected Segment root;
	
	public BaseSegment() {
		
	}
	
	public BaseSegment(RootSegment root) {
		super(-2);
		this.root = root;
	}
	
	public Segment findBranch(int word) {
		return root;
	}
	
}

File: stlm-java/src/segment/BaseSegment.py
#!/usr/bin/env python
""" generated source for module BaseSegment """
# package: segment
class BaseSegment(InternalSegment):
    """ generated source for class BaseSegment """
    root = Segment()

    @overloaded
    def __init__(self):
        """ generated source for method __init__ """
        super(BaseSegment, self).__init__()

    @__init__.register(object, RootSegment)
    def __init___0(self, root):
        """ generated source for method __init___0 """
        super(BaseSegment, self).__init__(-2)
        self.root = root

    def findBranch(self, word):
        """ generated source for method findBranch """
        return self.root

